pages/products/index.tsx

```diff
@@ -1,23 +1,38 @@
-import { Button, Dropdown, Panel, Small, StatefulTable, Link as StyledLink } from '@bigcommerce/big-design';
+import { Button, Dropdown, Panel, Small, Link as StyledLink, Table } from '@bigcommerce/big-design';
 import { MoreHorizIcon } from '@bigcommerce/big-design-icons';
 import Link from 'next/link';
 import { useRouter } from 'next/router';
-import { ReactElement } from 'react';
+import { ReactElement, useEffect, useState } from 'react';
 import ErrorMessage from '../../components/error';
 import Loading from '../../components/loading';
 import { useProductList } from '../../lib/hooks';
 import { TableItem } from '../../types';
 
 const Products = () => {
+    const [itemsPerPageOptions] = useState([10, 20, 50, 100]);
+    const [itemsPerPage, setItemsPerPage] = useState(10);
+    const [currentPage, setCurrentPage] = useState(1);
     const router = useRouter();
-    const { error, isLoading, list = [] } = useProductList();
+    const { error, isLoading, list = [], meta } = useProductList(currentPage, itemsPerPage);
     const tableItems: TableItem[] = list.map(({ id, inventory_level: stock, name, price }) => ({
         id,
         name,
         price,
         stock,
     }));
 
+    useEffect(() => {
+        if (list.length && meta) {
+            setCurrentPage(meta.pagination?.current_page);
+            setItemsPerPage(meta.pagination?.per_page);
+        }
+    }, [currentPage, list, meta]);
```

Comment:
rather than passing in objects (i.e. `list` and `meta`), we should destructure and pass in only the primitive values we need. useEffect, like most of JS, uses referential equality when comparing objects (i.e. in the dependency array). Essentially, this means that passing in an object will cause this useEffect to run every time